src/TraceEvent/TraceLog.cs

```diff
@@ -154,10 +154,192 @@ public static TraceLog OpenOrConvert(string etlOrEtlxFilePath, TraceLogOptions o
+        /// the .Log Property) which lets you get at aggregated information (Processes, threads, images loaded, and perhaps most
+        /// importantly TraceEvent.CallStack() will work. Thus you can get real time stacks from events).
+        /// </summary>
+        /// <param name="rundownConfiguration">
+        /// If enabled, a rundown is triggered immediately as a separate session.
+        /// This is used to initialize module and method information and then the session is closed.
+        /// This only makes sense in realtime sessions when you need to resolve function names during the session.
+        /// </param>
+        /// <example>
+        /// var client = new DiagnosticsClient(Process.GetCurrentProcess().Id);
+        /// EventPipeProvider[] providers = new[]
+        /// {   new EventPipeProvider(ClrTraceEventParser.ProviderName, EventLevel.Informational, (long) ClrTraceEventParser.Keywords.Default),
+        ///     new EventPipeProvider(SampleProfilerTraceEventParser.ProviderName, EventLevel.Informational),
+        /// };
+        /// var session = client.StartEventPipeSession(providers, requestRundown: false);
+        /// var eventSource = TraceLog.CreateFromEventPipeSession(session, TraceLog.EventPipeRundownConfiguration.Enable(client));
+        /// eventSource.Process();
+        /// </example>
+        public static TraceLogEventSource CreateFromEventPipeSession(EventPipeSession session, EventPipeRundownConfiguration rundownConfiguration = null)
+        {
+            var traceLog = new TraceLog(new EventPipeEventSource(session.EventStream));
+            traceLog.rawEventSourceToConvert.AllEvents += traceLog.OnAllEventPipeEventsRealTime;
+
+            var rundownDiagnosticsClient = rundownConfiguration?.m_client;
+            if (rundownDiagnosticsClient != null)
+            {
+                // Rundown events only come in after the session is stopped but we need them right from the start so that we
+                // can recognize loaded moodules and methods. Therefore, we start an additional session which will only collect
+                // rundown events and shut down immediately and feed this as an additional session to the TraceLog.
+                // Note: it doesn't matter what the actual provider is, just that we request rundown in the constructor.
+                using (var rundownSession = rundownDiagnosticsClient.StartEventPipeSession(
+                    new EventPipeProvider(ClrTraceEventParser.ProviderName, EventLevel.Informational, (long)ClrTraceEventParser.Keywords.Default),
+                    requestRundown: true
+                ))
+                {
+                    traceLog.ProcessInitialRundown(rundownSession);
+                }
+            }
+
             return traceLog.realTimeSource;
         }
 
+        public class EventPipeRundownConfiguration
```

Comment:
Please add comments to the public APIs here.

Reply:
docs added to the class & its public methods